packages/compiler/core/type-utils.ts

```diff
@@ -75,3 +79,50 @@ export function isGlobalNamespace(
 ): namespace is Namespace & { name: ""; namespace: undefined } {
   return program.getGlobalNamespaceType() === namespace;
 }
+
+/**
+ * Check if the given type is contained in the specified namespace or, optionally, its child namespaces.
+ * @param type Type
+ * @param namespace Namespace
+ * @returns {boolean}
+ */
+export function isDefinedInNamespace(
+  type: Model | Operation | Interface | Namespace | Enum,
+  namespace: Namespace,
+  options: { recursive?: boolean } = { recursive: true }
+) {
+  let collection: any[] = [];
+  switch (type.kind) {
+    case "Model":
+      collection = Array.from(namespace.models.values());
```

Comment:
Can't we traverse from bottom up instead of this(looking at the parent type)? This feels like this is quite expensive?